src/data/list/dedup.lean

```diff
@@ -48,6 +48,18 @@ dedup_cons_of_mem' $ mem_dedup.2 h
   dedup (a :: l) = a :: dedup l :=
 dedup_cons_of_not_mem' $ mt mem_dedup.1 h
 
+@[simp] lemma dedup_eq_nil (l : list α) : l.dedup = [] ↔ l = [] :=
+begin
+  induction l with a l hl,
+  { exact iff.rfl },
+  { by_cases h : a ∈ l,
+    { simp only [list.dedup_cons_of_mem h, hl, list.ne_nil_of_mem h] },
+    { simp only [list.dedup_cons_of_not_mem h, list.cons_ne_nil] } }
+end
+
+@[simp] lemma dedup_empty (l : list α) : l.dedup.empty ↔ l.empty :=
```

Comment:
I think this should be `empty_dedup`
```suggestion
@[simp] lemma empty_dedup {l : list α} : l.dedup.empty ↔ l.empty :=
```
(and arguments to `iff`s are usually implicit)

Reply:
Oh, actually `empty` is a `bool` so this should be
```suggestion
@[simp] lemma empty_dedup (l : list α) : l.dedup.empty = l.empty :=
```